### PR TITLE
Multiplayer code cleanup

### PR DESCRIPTION
**Pull request by GGGuenni**
- Fixed reviewed comments of PR 6273
- Moved Dropbox.kt from ui\worldscreen\mainmenu to logic\multiplayer
- Moved ServerMutex and OnlineMultiplayer to own files
- Changed DropBoxFileConflictException to FileStorageConflictException to match FileStorage scheme
- Merged FolderListEntry with DropboxMetaData as they were basically the same
- Introduced IFileMetaData to match FileStorage scheme and modified DropboxMetaData to use it
- Removed some unused imports

### CODE REVIEW

File: core/src/com/unciv/logic/multiplayer/DropBox.kt [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                println(ex.message)
+                val reader = BufferedReader(InputStreamReader(errorStream))
+                val responseString = reader.readText()
+                println(responseString)
+
+                // Throw Exceptions based on the HTTP response from dropbox
+                if (responseString.contains("path/not_found/"))
+                    throw FileNotFoundException()
+                if (responseString.contains("path/conflict/file"))
+                    throw FileStorageConflictException()
```
Comment (GGGuenni) [inline core/src/com/unciv/logic/multiplayer/DropBox.kt:47]: Marking everything that changed
- Changed DropBoxFileConflictException to FileStorageConflictException to match FileStorage scheme

File: core/src/com/unciv/logic/multiplayer/DropBox.kt [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+            return false
+        }
+    }
+
+    fun getFileMetaData(fileName: String): IFileMetaData {
+        val stream = dropboxApi("https://api.dropboxapi.com/2/files/get_metadata",
+                "{\"path\":\"$fileName\"}", "application/json")!!
+        val reader = BufferedReader(InputStreamReader(stream))
+        return GameSaver.json().fromJson(DropboxMetaData::class.java, reader.readText())
+    }
```
Comment (GGGuenni) [inline core/src/com/unciv/logic/multiplayer/DropBox.kt:119]: modified DropboxMetaData to use IFileMetaData

File: core/src/com/unciv/logic/multiplayer/DropBox.kt [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    @Suppress("PropertyName")
+    class DropboxMetaData: IFileMetaData {
+        var name = ""
+        private var server_modified = ""
+
+        override fun getFileName() = name
+        override fun getLastModified(): Date {
+            return server_modified.parseDate()
+        }
+    }
```
Comment (GGGuenni) [inline core/src/com/unciv/logic/multiplayer/DropBox.kt:145]: Merged FolderListEntry with DropboxMetaData as they were basically the same
Modified DropboxMetaData to use IFileMetaData

File: core/src/com/unciv/logic/multiplayer/Multiplayer.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    fun getFileMetaData(fileName: String): IFileMetaData
+    fun deleteFile(fileName: String)
+}
+
+interface IFileMetaData {
+    fun getFileName(): String?
+    fun getLastModified(): Date?
+}
+
+class FileStorageConflictException: Exception()
```
Comment (GGGuenni) [inline core/src/com/unciv/logic/multiplayer/Multiplayer.kt:21]: Introduced IFileMetaData to match FileStorage scheme
Changed DropBoxFileConflictException to FileStorageConflictException to match FileStorage scheme

File: core/src/com/unciv/logic/multiplayer/ServerMutex.kt [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+        locked = false
+        val fileName = "${gameInfo.gameId}_Lock"
+
+        // We have to check if the lock file already exists before we try to upload a new
+        // lock file to not overuse the dropbox file upload limit else it will return an error
+        try {
+            val metaData = OnlineMultiplayer().fileStorage.getFileMetaData(fileName)
+
+            val date = metaData.getLastModified()
```
Comment (GGGuenni) [inline core/src/com/unciv/logic/multiplayer/ServerMutex.kt:56]: Implemented usage for IFileMetaData

**Comment by SomeTroglodyte** [comment]
Still looks like a beginning. Lots on unused function / class / constructor / import etc, other inspection warnings...
DropboxMetaData is in the DropBox namespace anyway, so the long name is redundant, and doesn't match the comparable FolderList - which is only used by the unused getFolderList function.
<details><summary>Also, consider the IDE feature to give recognizable icons only to files dedicated to a single class/object/iface etc - can't at least a few of those be isolated meaningfully?</summary>

image
</details>

<details><summary>Just to pester and tease - <a href="https://github.com/SomeTroglodyte/Unciv/tree/cleanup">this branch</a> is derived from your commits.. as patch =</summary>

```patch
Index: core/src/com/unciv/logic/multiplayer/ServerMutex.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/src/com/unciv/logic/multiplayer/ServerMutex.kt b/core/src/com/unciv/logic/multiplayer/ServerMutex.kt
--- a/core/src/com/unciv/logic/multiplayer/ServerMutex.kt	(revision dc836b0f6d5dc06f8b7c70c14abbe7914ebe8840)
+++ b/core/src/com/unciv/logic/multiplayer/ServerMutex.kt	(date 1647021790352)
@@ -4,32 +4,20 @@
 import com.unciv.logic.GameInfoPreview
 import com.unciv.logic.GameSaver
 import com.unciv.ui.saves.Gzip
-import java.io.BufferedReader
 import java.io.FileNotFoundException
-import java.io.InputStreamReader
-import java.util.*
 import kotlin.math.pow
 
-/**
- * Used to communicate data access between players
- */
-class LockFile {
-    // The lockData is necessary to make every LockFile unique
-    // If Dropbox gets a file with the same content and overwrite set to false, it returns no
-    // error even though the file was not uploaded as the exact file is already existing
-    var lockData = UUID.randomUUID().toString()
-}
-
 /**
  *	Wrapper around OnlineMultiplayer's synchronization facilities.
  *
  *	Based on the design of Mutex from kotlinx.coroutines.sync, except that when it blocks,
  *	  it blocks the entire thread for an increasing period of time via Thread.sleep()
  */
+@Suppress("unused")   // Work in progress, will use later
 class ServerMutex(val gameInfo: GameInfoPreview) {
     private var locked = false
 
-    constructor(gameInfo: GameInfo) : this(gameInfo.asPreview()) { }
+    constructor(gameInfo: GameInfo) : this(gameInfo.asPreview())
 
     /**
      * Try to obtain the server lock ONCE
Index: core/src/com/unciv/logic/multiplayer/DropBox.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/src/com/unciv/logic/multiplayer/DropBox.kt b/core/src/com/unciv/logic/multiplayer/DropboxFileStorage.kt
rename from core/src/com/unciv/logic/multiplayer/DropBox.kt
rename to core/src/com/unciv/logic/multiplayer/DropboxFileStorage.kt
--- a/core/src/com/unciv/logic/multiplayer/DropBox.kt	(revision dc836b0f6d5dc06f8b7c70c14abbe7914ebe8840)
+++ b/core/src/com/unciv/logic/multiplayer/DropboxFileStorage.kt	(date 1647022320470)
@@ -7,163 +7,169 @@
 import java.net.URL
 import java.nio.charset.Charset
 import java.util.*
-import kotlin.collections.ArrayList
+
 
+class DropboxFileStorage: IFileStorage {
+    // This is the location in Dropbox only
+    private fun getLocalGameLocation(fileName: String) = "/MultiplayerGames/$fileName"
 
-object DropBox {
-    fun dropboxApi(url: String, data: String = "", contentType: String = "", dropboxApiArg: String = ""): InputStream? {
+    override fun saveFileData(fileName: String, data: String) {
+        val fileLocationDropbox = getLocalGameLocation(fileName)
+        DropBox.uploadFile(fileLocationDropbox, data, true)
+    }
+
+    override fun loadFileData(fileName: String): String {
+        return DropBox.downloadFileAsString(getLocalGameLocation(fileName))
+    }
+
+    override fun getFileMetaData(fileName: String): IFileMetaData {
+        return DropBox.getFileMetaData(getLocalGameLocation(fileName))
+    }
+
+    override fun deleteFile(fileName: String) {
+        DropBox.deleteFile(getLocalGameLocation(fileName))
+    }
+
+    private object DropBox {
+        private fun dropboxApi(url: String, data: String = "", contentType: String = "", dropboxApiArg: String = ""): InputStream? {
 
-        with(URL(url).openConnection() as HttpURLConnection) {
-            requestMethod = "POST"  // default is GET
+            with(URL(url).openConnection() as HttpURLConnection) {
+                requestMethod = "POST"  // default is GET
 
-            @Suppress("SpellCheckingInspection")
-            setRequestProperty("Authorization", "Bearer LTdBbopPUQ0AAAAAAAACxh4_Qd1eVMM7IBK3ULV3BgxzWZDMfhmgFbuUNF_rXQWb")
+                @Suppress("SpellCheckingInspection")
+                setRequestProperty("Authorization", "Bearer LTdBbopPUQ0AAAAAAAACxh4_Qd1eVMM7IBK3ULV3BgxzWZDMfhmgFbuUNF_rXQWb")
 
-            if (dropboxApiArg != "") setRequestProperty("Dropbox-API-Arg", dropboxApiArg)
-            if (contentType != "") setRequestProperty("Content-Type", contentType)
+                if (dropboxApiArg != "") setRequestProperty("Dropbox-API-Arg", dropboxApiArg)
+                if (contentType != "") setRequestProperty("Content-Type", contentType)
 
-            doOutput = true
+                doOutput = true
 
-            try {
-                if (data != "") {
-                    // StandardCharsets.UTF_8 requires API 19
-                    val postData: ByteArray = data.toByteArray(Charset.forName("UTF-8"))
-                    val outputStream = DataOutputStream(outputStream)
-                    outputStream.write(postData)
-                    outputStream.flush()
-                }
+                try {
+                    if (data != "") {
+                        // StandardCharsets.UTF_8 requires API 19
+                        val postData: ByteArray = data.toByteArray(Charset.forName("UTF-8"))
+                        val outputStream = DataOutputStream(outputStream)
+                        outputStream.write(postData)
+                        outputStream.flush()
+                    }
 
-                return inputStream
-            } catch (ex: Exception) {
-                println(ex.message)
-                val reader = BufferedReader(InputStreamReader(errorStream))
-                val responseString = reader.readText()
-                println(responseString)
+                    return inputStream
+                } catch (ex: Exception) {
+                    println(ex.message)
+                    val reader = BufferedReader(InputStreamReader(errorStream))
+                    val responseString = reader.readText()
+                    println(responseString)
 
-                // Throw Exceptions based on the HTTP response from dropbox
-                if (responseString.contains("path/not_found/"))
-                    throw FileNotFoundException()
-                if (responseString.contains("path/conflict/file"))
-                    throw FileStorageConflictException()
-                
-                return null
-            } catch (error: Error) {
-                println(error.message)
-                val reader = BufferedReader(InputStreamReader(errorStream))
-                println(reader.readText())
-                return null
-            }
-        }
-    }
+                    // Throw Exceptions based on the HTTP response from dropbox
+                    if (responseString.contains("path/not_found/"))
+                        throw FileNotFoundException()
+                    if (responseString.contains("path/conflict/file"))
+                        throw FileStorageConflictException()
+
+                    return null
+                } catch (error: Error) {
+                    println(error.message)
+                    val reader = BufferedReader(InputStreamReader(errorStream))
+                    println(reader.readText())
+                    return null
+                }
+            }
+        }
 
-    fun getFolderList(folder: String): ArrayList<DropboxMetaData> {
-        val folderList = ArrayList<DropboxMetaData>()
-        // The DropBox API returns only partial file listings from one request. list_folder and
-        // list_folder/continue return similar responses, but list_folder/continue requires a cursor
-        // instead of the path.
-        val response = dropboxApi("https://api.dropboxapi.com/2/files/list_folder",
+/* unused - but keep it for future designs
+        fun getFolderList(folder: String): ArrayList<DropboxMetaData> {
+            val folderList = ArrayList<DropboxMetaData>()
+            // The DropBox API returns only partial file listings from one request. list_folder and
+            // list_folder/continue return similar responses, but list_folder/continue requires a cursor
+            // instead of the path.
+            val response = dropboxApi("https://api.dropboxapi.com/2/files/list_folder",
                 "{\"path\":\"$folder\"}", "application/json")
-        var currentFolderListChunk = GameSaver.json().fromJson(FolderList::class.java, response)
-        folderList.addAll(currentFolderListChunk.entries)
-        while (currentFolderListChunk.has_more) {
-            val continuationResponse = dropboxApi("https://api.dropboxapi.com/2/files/list_folder/continue",
+            var currentFolderListChunk = GameSaver.json().fromJson(FolderList::class.java, response)
+            folderList.addAll(currentFolderListChunk.entries)
+            while (currentFolderListChunk.has_more) {
+                val continuationResponse = dropboxApi("https://api.dropboxapi.com/2/files/list_folder/continue",
                     "{\"cursor\":\"${currentFolderListChunk.cursor}\"}", "application/json")
-            currentFolderListChunk = GameSaver.json().fromJson(FolderList::class.java, continuationResponse)
-            folderList.addAll(currentFolderListChunk.entries)
-        }
-        return folderList
-    }
+                currentFolderListChunk = GameSaver.json().fromJson(FolderList::class.java, continuationResponse)
+                folderList.addAll(currentFolderListChunk.entries)
+            }
+            return folderList
+        }
+*/
 
-    fun downloadFile(fileName: String): InputStream {
-        val response = dropboxApi("https://content.dropboxapi.com/2/files/download",
+        fun downloadFile(fileName: String): InputStream {
+            val response = dropboxApi("https://content.dropboxapi.com/2/files/download",
                 contentType = "text/plain", dropboxApiArg = "{\"path\":\"$fileName\"}")
-        return response!!
-    }
+            return response!!
+        }
 
-    fun downloadFileAsString(fileName: String): String {
-        val inputStream = downloadFile(fileName)
-        return BufferedReader(InputStreamReader(inputStream)).readText()
-    }
+        fun downloadFileAsString(fileName: String): String {
+            val inputStream = downloadFile(fileName)
+            return BufferedReader(InputStreamReader(inputStream)).readText()
+        }
 
-    /**
-     * @param overwrite set to true to avoid DropBoxFileConflictException
-     * @throws DropBoxFileConflictException when overwrite is false and a file with the
-     * same name already exists
-     */
-    fun uploadFile(fileName: String, data: String, overwrite: Boolean = false) {
-        val overwriteModeString = if(!overwrite) "" else ""","mode":{".tag":"overwrite"}"""
-        dropboxApi("https://content.dropboxapi.com/2/files/upload",
+        /**
+         * @param overwrite set to true to avoid DropBoxFileConflictException
+         * @throws FileStorageConflictException when overwrite is false and a file with the
+         * same name already exists
+         */
+        fun uploadFile(fileName: String, data: String, overwrite: Boolean = false) {
+            val overwriteModeString = if(!overwrite) "" else ""","mode":{".tag":"overwrite"}"""
+            dropboxApi("https://content.dropboxapi.com/2/files/upload",
                 data, "application/octet-stream", """{"path":"$fileName"$overwriteModeString}""")
-    }
+        }
 
-    fun deleteFile(fileName: String){
-        dropboxApi("https://api.dropboxapi.com/2/files/delete_v2",
+        fun deleteFile(fileName: String){
+            dropboxApi("https://api.dropboxapi.com/2/files/delete_v2",
                 "{\"path\":\"$fileName\"}", "application/json")
-    }
+        }
 
-    fun fileExists(fileName: String): Boolean {
-        try {
-            dropboxApi("https://api.dropboxapi.com/2/files/get_metadata",
+/* unused - but keep it for future designs
+        fun fileExists(fileName: String): Boolean {
+            return try {
+                dropboxApi("https://api.dropboxapi.com/2/files/get_metadata",
                     "{\"path\":\"$fileName\"}", "application/json")
-            return true
-        } catch (ex: FileNotFoundException) {
-            return false
-        }
-    }
+                true
+            } catch (ex: FileNotFoundException) {
+                false
+            }
+        }
+*/
 
-    fun getFileMetaData(fileName: String): IFileMetaData {
-        val stream = dropboxApi("https://api.dropboxapi.com/2/files/get_metadata",
+        fun getFileMetaData(fileName: String): IFileMetaData {
+            val stream = dropboxApi("https://api.dropboxapi.com/2/files/get_metadata",
                 "{\"path\":\"$fileName\"}", "application/json")!!
-        val reader = BufferedReader(InputStreamReader(stream))
-        return GameSaver.json().fromJson(DropboxMetaData::class.java, reader.readText())
-    }
+            val reader = BufferedReader(InputStreamReader(stream))
+            return GameSaver.json().fromJson(DropboxMetaData::class.java, reader.readText())
+        }
 
-//
-//    fun createTemplate(): String {
-//        val result =  dropboxApi("https://api.dropboxapi.com/2/file_properties/templates/add_for_user",
-//                "{\"name\": \"Security\",\"description\": \"These properties describe how confidential this file or folder is.\",\"fields\": [{\"name\": \"Security Policy\",\"description\": \"This is the security policy of the file or folder described.\nPolicies can be Confidential, Public or Internal.\",\"type\": \"string\"}]}"
-//                ,"application/json")
-//        return BufferedReader(InputStreamReader(result)).readText()
-//    }
 
-    @Suppress("PropertyName")
-    class FolderList{
-        var entries = ArrayList<DropboxMetaData>()
-        var cursor = ""
-        var has_more = false
-    }
+/* unused - but keep it for future designs
+        fun createTemplate(): String {
+            val result =  dropboxApi("https://api.dropboxapi.com/2/file_properties/templates/add_for_user",
+                    "{\"name\": \"Security\",\"description\": \"These properties describe how confidential this file or folder is.\",\"fields\": [{\"name\": \"Security Policy\",\"description\": \"This is the security policy of the file or folder described.\nPolicies can be Confidential, Public or Internal.\",\"type\": \"string\"}]}"
+                    ,"application/json")
+            return BufferedReader(InputStreamReader(result)).readText()
+        }
+*/
+
+/* unused - but keep it for future designs
+        @Suppress("PropertyName")
+        class FolderList{
+            var entries = ArrayList<DropboxMetaData>()
+            var cursor = ""
+            var has_more = false
+        }
+*/
 
-    @Suppress("PropertyName")
-    class DropboxMetaData: IFileMetaData {
-        var name = ""
-        private var server_modified = ""
+        @Suppress("PropertyName", "PrivatePropertyName")
+        class DropboxMetaData: IFileMetaData {
+            var name = ""
+            private var server_modified = ""
 
-        override fun getFileName() = name
-        override fun getLastModified(): Date {
-            return server_modified.parseDate()
-        }
-    }
-}
-
-class DropboxFileStorage: IFileStorage {
-    // This is the location in Dropbox only
-    fun getLocalGameLocation(fileName: String) = "/MultiplayerGames/$fileName"
-
-    override fun saveFileData(fileName: String, data: String) {
-        val fileLocationDropbox = getLocalGameLocation(fileName)
-        DropBox.uploadFile(fileLocationDropbox, data, true)
+            override fun getFileName() = name
+            override fun getLastModified(): Date {
+                return server_modified.parseDate()
+            }
+        }
     }
-
-    override fun loadFileData(fileName: String): String {
-        return DropBox.downloadFileAsString(getLocalGameLocation(fileName))
-    }
-
-    override fun getFileMetaData(fileName: String): IFileMetaData {
-        return DropBox.getFileMetaData(getLocalGameLocation(fileName))
-    }
-
-    override fun deleteFile(fileName: String) {
-        DropBox.deleteFile(getLocalGameLocation(fileName))
-    }
-
 }
Index: core/src/com/unciv/logic/multiplayer/Multiplayer.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/src/com/unciv/logic/multiplayer/Multiplayer.kt b/core/src/com/unciv/logic/multiplayer/OnlineMultiplayer.kt
rename from core/src/com/unciv/logic/multiplayer/Multiplayer.kt
rename to core/src/com/unciv/logic/multiplayer/OnlineMultiplayer.kt
--- a/core/src/com/unciv/logic/multiplayer/Multiplayer.kt	(revision dc836b0f6d5dc06f8b7c70c14abbe7914ebe8840)
+++ b/core/src/com/unciv/logic/multiplayer/OnlineMultiplayer.kt	(date 1647022320498)
@@ -4,24 +4,9 @@
 import com.unciv.logic.GameInfoPreview
 import com.unciv.logic.GameSaver
 import com.unciv.ui.saves.Gzip
-import java.util.*
-
-interface IFileStorage {
-    fun saveFileData(fileName: String, data: String)
-    fun loadFileData(fileName: String): String
-    fun getFileMetaData(fileName: String): IFileMetaData
-    fun deleteFile(fileName: String)
-}
 
-interface IFileMetaData {
-    fun getFileName(): String?
-    fun getLastModified(): Date?
-}
-
-class FileStorageConflictException: Exception()
-
-class OnlineMultiplayer {
-    val fileStorage: IFileStorage = DropboxFileStorage()
+class OnlineMultiplayer(type: IFileStorage.Type = IFileStorage.Type.Dropbox) {
+    val fileStorage: IFileStorage = IFileStorage.getStorageProvider(type)
 
     fun tryUploadGame(gameInfo: GameInfo, withPreview: Boolean) {
         // We upload the gamePreview before we upload the game as this
@@ -40,6 +25,7 @@
      * @see tryUploadGame
      * @see GameInfo.asPreview
      */
+    @Suppress("MemberVisibilityCanBePrivate")  // Meant to be callable w/o tryUploadGame
     fun tryUploadGamePreview(gameInfo: GameInfoPreview) {
         val zippedGameInfo = Gzip.zip(GameSaver.json().toJson(gameInfo))
         fileStorage.saveFileData("${gameInfo.gameId}_Preview", zippedGameInfo)
Index: core/src/com/unciv/logic/multiplayer/UNCPathFileStorage.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/src/com/unciv/logic/multiplayer/UNCPathFileStorage.kt b/core/src/com/unciv/logic/multiplayer/UNCPathFileStorage.kt
new file mode 100644
--- /dev/null	(date 1647022320510)
+++ b/core/src/com/unciv/logic/multiplayer/UNCPathFileStorage.kt	(date 1647022320510)
@@ -0,0 +1,19 @@
+package com.unciv.logic.multiplayer
+
+class UNCPathFileStorage: IFileStorage {
+    override fun saveFileData(fileName: String, data: String) {
+        TODO("Not yet implemented")
+    }
+
+    override fun loadFileData(fileName: String): String {
+        TODO("Not yet implemented")
+    }
+
+    override fun getFileMetaData(fileName: String): IFileMetaData {
+        TODO("Not yet implemented")
+    }
+
+    override fun deleteFile(fileName: String) {
+        TODO("Not yet implemented")
+    }
+}
\ No newline at end of file
Index: core/src/com/unciv/logic/multiplayer/MultiplayerHelpers.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/src/com/unciv/logic/multiplayer/MultiplayerHelpers.kt b/core/src/com/unciv/logic/multiplayer/MultiplayerHelpers.kt
new file mode 100644
--- /dev/null	(date 1647022320486)
+++ b/core/src/com/unciv/logic/multiplayer/MultiplayerHelpers.kt	(date 1647022320486)
@@ -0,0 +1,43 @@
+package com.unciv.logic.multiplayer
+
+import java.util.*
+
+interface IFileStorage {
+    fun saveFileData(fileName: String, data: String)
+    fun loadFileData(fileName: String): String
+    fun getFileMetaData(fileName: String): IFileMetaData
+    fun deleteFile(fileName: String)
+
+    enum class Type {
+        Dropbox,
+        Webdav,
+        UNCPath,
+    }
+
+    companion object {
+        fun getStorageProvider(type: Type): IFileStorage = when(type) {
+            Type.Dropbox -> DropboxFileStorage()
+            Type.Webdav -> WebdavFileStorage()
+            Type.UNCPath -> UNCPathFileStorage()
+        }
+    }
+}
+
+interface IFileMetaData {
+    fun getFileName(): String?
+    fun getLastModified(): Date?
+}
+
+class FileStorageConflictException: Exception()
+
+/**
+ * Used to communicate data access between players
+ */
+class LockFile {
+    // The lockData is necessary to make every LockFile unique
+    // If Dropbox gets a file with the same content and overwrite set to false, it returns no
+    // error even though the file was not uploaded as the exact file is already existing
+    @Suppress("unused")  // used through serialization
+    var lockData = UUID.randomUUID().toString()
+}
+
Index: core/src/com/unciv/logic/multiplayer/WebdavFileStorage.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/src/com/unciv/logic/multiplayer/WebdavFileStorage.kt b/core/src/com/unciv/logic/multiplayer/WebdavFileStorage.kt
new file mode 100644
--- /dev/null	(date 1647021563942)
+++ b/core/src/com/unciv/logic/multiplayer/WebdavFileStorage.kt	(date 1647021563942)
@@ -0,0 +1,19 @@
+package com.unciv.logic.multiplayer
+
+class WebdavFileStorage: IFileStorage {
+    override fun saveFileData(fileName: String, data: String) {
+        TODO("Not yet implemented")
+    }
+
+    override fun loadFileData(fileName: String): String {
+        TODO("Not yet implemented")
+    }
+
+    override fun getFileMetaData(fileName: String): IFileMetaData {
+        TODO("Not yet implemented")
+    }
+
+    override fun deleteFile(fileName: String) {
+        TODO("Not yet implemented")
+    }
+}
\ No newline at end of file
```
</details>

... food for thought as always.

**Comment by GGGuenni** [comment]
Holy cow! Wasn't expecting a whole patch for my sloppy code! Thanks for your work!

> Still looks like a beginning.

Definitely! Wasn't really meant to be more than that. Just couldn't stand putting everything new I add on other branches into the same dropbox file anymore 😅

> Also, consider the IDE feature to give recognizable icons only to files dedicated to a single class/object/iface etc - can't at least a few of those be isolated meaningfully?  

I choose the file icon as all files contained more than one class. With your patch, it makes sense to change icons

> Just to pester and tease - [this branch](https://github.com/SomeTroglodyte/Unciv/tree/cleanup) is derived from your commits..

Just create a PR on my branch so I can merge it and keep your work labeled as such. Always happy to combine work.

**Comment by SomeTroglodyte** [comment]
> Just create a PR on my branch

Nah, too lazy to find out how. That was just 15min add your remote, checkout, shuffle around until "Problems" window (Alt-6) empty for those files. And the patch won't compile as it made the downloader function inaccesible to DL the thatched villagers soundtrack... The branch with a second commit runs, however. Take it as an _idea_ where that could go. The most clever decision was how to make the `object DropBox` and the class to a file with a class icon - object to companion, all members private except what music needs was third attempt or so. In fact almost all the statics in object DropBox could live as class methods, but I was lazy.

**Pull request merged**